### PR TITLE
fix: regtest network mode

### PR DESCRIPTION
**Pull request by fbwoolf**
## Description

This removes regtest as a default network from the explorer.

For details refer to issue #539 

Screen Shot 2021-09-15 at 1 52 06 PM

## Type of Change
- [ ] New feature
- [x] Bug fix
- [ ] API reference/documentation update
- [ ] Other

## Testing information

- Make sure switching to/from regtest network works properly with toast, url chain, and dropdown list

### CODE REVIEW

File: src/common/api/utils.ts [hunk 1]
```diff
@@ -121,6 +129,8 @@ export const getChainTypeFromUrl = (networkUrl: string | undefined) => {
       return NetworkModes.Mainnet;
     case DEFAULT_TESTNET_SERVER:
       return NetworkModes.Testnet;
+    case DEFAULT_REGTEST_SERVER:
+      return NetworkModes.Regtest;
```
Comment (fbwoolf) [inline src/common/api/utils.ts:133]: This missing seemed to be the real issue ...the chain came in `undefined`.

**Review by kyranjamie — CHANGES_REQUESTED** [review]
As I understand, there is no "regtest" chain mode. There is only `mainnet` and `testnet`. Regtest is of chain mode testnet (referred to as `ChainID` in stacks.js).

It's confusing because we refer to the networks with the same name, but there's distinction between the two which isn't reflected here.

Let's talk it through later, but I think we want some kind of map bridging network to chainid. Something like:

```
{
  mainnet: ChainId.Mainnet,
  testnet: ChainId.Testnet,
  regtest: ChainId.Testnet,
}
```

File: src/common/api/utils.ts (hunk 1 above)
Comment (kyranjamie) [reply src/common/api/utils.ts:133]: Shouldn't we return `NetworkMode.Testnet` here?

**Comment by fbwoolf** [comment]
> 
> Let's talk it through later, but I think we want some kind of map bridging network to chainid. Something like:
> 

I think maybe the explorer has it setup a bit differently client-side bc there is a `NetworkModes.Regtest`
https://github.com/blockstack/explorer/blob/a1598bc6eca597c32b3b428c8fdd77c09981a8da/src/common/types/network.ts

The first thing the app does is get the `apiServer`, so the regtest url, `DEFAULT_REGTEST_SERVER` is used to then get that `NetworkMode` and use it. I think what you are suggesting would be refactoring how things have already been setup rather than just fixing this issue?

**Comment by aulneau** [comment]
One thing to note -- it's very likely that regtest will be decommissioned, I think we should probably remove it from the explorer.

**Comment by lgalabru** [comment]
Should we just get rid of the Regtest option, and replace it with Devnet?

**Comment by aulneau** [comment]
> Should we just get rid of the Regtest option, and replace it with Devnet?

Devnet is a local only version, right? local (non-https) urls will only work if the explorer is running locally, too.

**Comment by markmhx** [comment]
Related: https://github.com/hirosystems/devops/issues/763

cc @CharlieC3

**Comment by markmhx** [comment]
> Should we just get rid of the Regtest option, and replace it with Devnet?

I'd suggest this ☝️

**Comment by fbwoolf** [comment]
@kyranjamie can you circle back and take a look through this one again?

**Review by kyranjamie — APPROVED** [review]
Circling back 🌀

**Comment by LunarPunks** [comment]
Guys where are the settings in local explorer to make devnet the default,  i think this causing problems related to issue 647 and 648 ,  All actions divert to mainnet,  I need access to change this,  cannot do any local sandbox and explorer activities.

my setup is windows 10, clarinet, docker -  contracts deploy fine from clarinet --integrate but we cannot interact with them in sandbox,  because of the above mentioned problem.

**Pull request merged**